plugin/gitsessions.vim

```diff
@@ -65,37 +65,45 @@ function! s:os_sep()
 endfunction
 
 function! s:is_abs_path(path)
-    return a:path[0] == s:os_sep()
+    if has('unix')
+        return a:path[0] == s:os_sep()
+    else
+        return a:path[1] == ':'
+    endif
 endfunction
 
 " LOGIC FUNCTIONS
 
 function! s:parent_dir(path)
     let l:sep = s:os_sep()
-    let l:front = s:is_abs_path(a:path) ? l:sep : ''
+    let l:front = s:is_abs_path(a:path) && has('unix') ? l:sep : ''
     return l:front . join(split(a:path, l:sep)[:-2], l:sep)
 endfunction
 
 function! s:find_git_dir(dir)
-    if isdirectory(a:dir . '/.git')
-        return a:dir . '/.git'
-    elseif has('file_in_path') && has('path_extra')
+    if isdirectory(a:dir . s:os_sep() . '.git')
+        return a:dir . s:os_sep() . '.git'
+    elseif has('file_in_path') && has('path_extra') && has('unix')
         return finddir('.git', a:dir . ';')
     else
         return s:find_git_dir_aux(a:dir)
     endif
 endfunction
 
 function! s:find_git_dir_aux(dir)
-    return isdirectory(a:dir . '/.git') ? a:dir . '/.git' : s:find_git_dir_aux(s:parent_dir(a:dir))
+    return isdirectory(a:dir . s:os_sep() . '.git') ? a:dir . s:os_sep() . '.git' : s:find_git_dir_aux(s:parent_dir(a:dir))
 endfunction
 
 function! s:find_project_dir(dir)
     return s:parent_dir(s:find_git_dir(a:dir))
 endfunction
 
 function! s:session_path(sdir, pdir)
-    let l:path = a:sdir . a:pdir
+    if has('unix')
+        let l:path = a:sdir . a:pdir
+    else
+        let l:path = a:sdir . substitute(a:pdir,'^\([A-Z]\):','\\\1','g')
```

Comment:
This is some regex backreferencing stuff. Can you help me understand why this change is necessary?